From 3697446e5b9897c5189d8ea1a98e093e3a9c6d89 Mon Sep 17 00:00:00 2001
From: Sam Nazarko <email@samnazarko.co.uk>
Date: Fri, 10 Apr 2015 18:06:32 +0100
Subject: [PATCH] iMX: Only align dest Rects to 2 - cares for artifacts

* Fixes artifacts when aligning videos that are not power of 8
---
 .../dvdplayer/DVDCodecs/Video/DVDVideoCodecIMX.cpp     | 18 +++++++++---------
 1 file changed, 9 insertions(+), 9 deletions(-)

diff --git a/xbmc/cores/dvdplayer/DVDCodecs/Video/DVDVideoCodecIMX.cpp b/xbmc/cores/dvdplayer/DVDCodecs/Video/DVDVideoCodecIMX.cpp
index 4c56f51..dee1ad4 100644
--- a/xbmc/cores/dvdplayer/DVDCodecs/Video/DVDVideoCodecIMX.cpp
+++ b/xbmc/cores/dvdplayer/DVDCodecs/Video/DVDVideoCodecIMX.cpp
@@ -1784,15 +1784,15 @@ void CIMXContext::PrepareTask(IPUTask &ipu, CIMXBuffer *source_p, CIMXBuffer *so
     dstRect.y2 = m_fbHeight;
   }
 
-  iSrcRect.x1 = Align((int)srcRect.x1,8);
-  iSrcRect.y1 = Align((int)srcRect.y1,8);
-  iSrcRect.x2 = Align((int)srcRect.x2,8);
-  iSrcRect.y2 = Align((int)srcRect.y2,8);
-
-  iDstRect.x1 = Align((int)dstRect.x1,8);
-  iDstRect.y1 = Align((int)dstRect.y1,8);
-  iDstRect.x2 = Align((int)dstRect.x2,8);
-  iDstRect.y2 = Align((int)dstRect.y2,8);
+  iSrcRect.x1 = Align((int)srcRect.x1,2);
+  iSrcRect.y1 = Align((int)srcRect.y1,2);
+  iSrcRect.x2 = Align((int)srcRect.x2,2);
+  iSrcRect.y2 = Align((int)srcRect.y2,2);
+
+  iDstRect.x1 = Align((int)dstRect.x1,2);
+  iDstRect.y1 = Align((int)dstRect.y1,2);
+  iDstRect.x2 = Align((int)dstRect.x2,2);
+  iDstRect.y2 = Align((int)dstRect.y2,2);
 
   ipu.task.input.crop.pos.x  = iSrcRect.x1;
   ipu.task.input.crop.pos.y  = iSrcRect.y1;
